redwood-layout-uiview/src/commonMain/kotlin/app/cash/redwood/layout/uiview/RedwoodUIScrollViewFactory.kt

```diff
@@ -15,30 +15,17 @@
  */
 package app.cash.redwood.layout.uiview
 
-import app.cash.redwood.flexbox.Size
+import app.cash.redwood.layout.uiview.cinterop.HostView
+import app.cash.redwood.layout.uiview.cinterop.RedwoodScrollViewDelegateProtocol
 import platform.UIKit.UIScrollView
 
-/**
- * A delegate whose methods should be called when the corresponding [UIScrollView] method is invoked.
- */
-public interface RedwoodUIScrollViewDelegate {
-  public val intrinsicContentSize: UnsafeSize
-  public fun sizeThatFits(size: UnsafeSize): UnsafeSize
-  public fun setNeedsLayout()
-  public fun layoutSubviews()
-}
-
 /**
  * A factory that creates [UIScrollView] subclasses that delegate to a [RedwoodUIScrollViewDelegate].
  */
-public interface RedwoodUIScrollViewFactory {
-  public fun create(delegate: RedwoodUIScrollViewDelegate): UIScrollView
+public class RedwoodUIScrollViewFactory {
```

Comment:
We can remove this too and allocate the `HostView` directly inside `UIViewFlexContainer`.

Reply:
Updated